app/controllers/posts_controller.rb

```diff
@@ -27,6 +27,18 @@ def create
     end
   end
 
+  def destroy
+    @user = current_user
+    @post = @user.posts.find_by(id: params[:id])
+
+    puts @post.inspect
+    if @post.destroy
+      redirect_to user_posts_path(@user), notice: 'Post deleted successfully!'
+    else
+      redirect_to user_posts_path(@user), alert: 'Failed to delete the post.'
+    end
+  end
+
```

Comment:
- [ ] I also noticed that in your `posts_controller`, it does not look like the CanCanCan gem is being used to handle post-authorization. There is no reference to `load_and_authorize_resource` or any other CanCanCan-related methods in your code. So kindly fix this :+1:
- You can simply Load and authorize resources in the PostsController as shown below :+1: 
```ruby
class PostsController < ApplicationController
  load_and_authorize_resource

  # ... other controller methods ...
end
```